### PR TITLE
fix: refactor BlobResponse and ArrayBufferResponse

### LINKED ISSUE #11: StreamBlobResponse should not return a promise
I'm working on integrating this polyfill to enable text streaming in my app and have encountered a problem that goes like this:
```js
fetch('...').then(response => {
    console.log(response.headers); // map object
    const clone = response.clone();
    console.log(clone.headers); // undefined, and headers isn't optional
    clone.then(cloneResponse => {
        console.log(cloneResponse.headers); // map object
    })
})
```

It seems like StreamBlobResponse constructor should not return a promise https://github.com/react-native-community/fetch/blob/master/src/StreamBlobResponse.js#L11-L22

I experimented with removing the returns and that does seem to fix it. The implementation treats the respective class properties as optional so everything continues to work. Was this an oversight or is there a deeper reason that it's implemented like this which breaks the Response object?

### PR DESCRIPTION
**Pull request by slightlytyler**
closes #11

simplifies implementation of both classes by extending `Request` and only supplying `constructor` and `clone`. Added some tests that are copy of existing tests with a `clone` call shoved in there. Haven't updated `package.json` or changelog yet

### CODE REVIEW

**Comment by acostalima** [comment]
@slightlytyler are you done?

**Comment by slightlytyler** [comment]
@acostalima yeah man I think this is g2g. all tests pass for me including the new initial failing specs i added

**Comment by acostalima** [comment]
@slightlytyler could you please make this a breaking change in the commit message? Although the `clone` method was not implemented correctly as per spec, it is a breaking change anyway.

**Comment by slightlytyler** [comment]
yep!

**Comment by slightlytyler** [comment]
@acostalima package version bumped, changelog updates. think this should be ready

File: package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "react-native-fetch-api",
   "description": "A fetch API polyfill for React Native with text streaming support.",
-  "version": "2.0.0",
+  "version": "3.0.0",
```
Comment (acostalima) [inline package.json:4]: Please revert this change. The version is bumped automatically during the release process.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,12 @@
 
 All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.
 
+## [3.0.0](https://github.com/react-native-community/fetch/compare/v2.0.0...v3.0.0) (2021-08-02)
+
+### Bug Fixes
+
+* do not return a promise from BlobResponseConstructor ([3745808](https://github.com/react-native-community/fetch/pull/12/commits/37458086f22e04b11e7fb2bb5e701999f1def02e))
+
```
Comment (acostalima) [inline CHANGELOG.md:10]: Please revert this change. The changelog is updated automatically during the release process.

**Review by acostalima — CHANGES_REQUESTED** [review]

**Comment by acostalima** [comment]
Confer https://www.conventionalcommits.org/en/v1.0.0/#examples to how a breaking change is done via commit message.

File: CHANGELOG.md (hunk 2 above)
Comment (slightlytyler) [reply CHANGELOG.md:10]: aha

**Comment by slightlytyler** [comment]
lol i cant read changing

**Comment by slightlytyler** [comment]
@acostalima should be good now, thanks for the hand holding

**Review by acostalima — APPROVED** [review]
